packages/kit/src/runtime/server/page/actions.js

```diff
@@ -0,0 +1,179 @@
+import { error, json } from '../../../exports/index.js';
+import { normalize_error } from '../../../utils/error.js';
+import { negotiate } from '../../../utils/http.js';
+import { HttpError, Redirect, ValidationError } from '../../control.js';
+import { error_to_pojo } from '../utils.js';
+
+// Info: `enhance` action and `updateForm` live in `runtime/app/forms.js`
+
+/** @param {import('types').RequestEvent} event */
+export function is_action_json_request(event) {
+	const accept = negotiate(event.request.headers.get('accept') || 'text/html', [
+		'text/html',
+		'application/json'
+	]);
+
+	return (
+		accept === 'application/json' &&
+		event.request.method !== 'GET' &&
+		event.request.method !== 'HEAD'
```

Comment:
the joys of getting rid of method overrides

```suggestion
		event.request.method === 'POST'
```